### PR TITLE
Range update of an array added with tests

### PR DESCRIPTION
**Pull request by AtharvaThorve**
Issue #150 resolved.
Algorithm for range update of an array using difference array added.
Tests are also added.
I've added only 2 files `RangeUpdate.java` and `RangeUpdateTest.java`. All other changes happened after executing `gradle goJF`.

### CODE REVIEW

File: src/test/java/com/williamfiset/algorithms/other/RangeUpdateTest.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package com.williamfiset.algorithms.other;
+
+import static org.junit.Assert.*;
+
+import org.junit.Test;
+
```
Comment (williamfiset) [inline src/test/java/com/williamfiset/algorithms/other/RangeUpdateTest.java:6]: Thanks for adding tests!

File: src/main/java/com/williamfiset/algorithms/other/RangeUpdate.java [hunk 2]
```diff
@@ -9,10 +10,19 @@
 
 public class RangeUpdate {
 
+  private int[] array;
+  private int[] differenceArray;
```
Comment (AtharvaThorve) [inline src/main/java/com/williamfiset/algorithms/other/RangeUpdate.java:14]: I have internalized the arrays so the user never has to pass them as arguments to update

File: src/main/java/com/williamfiset/algorithms/other/RangeUpdate.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    int n = array.length;
+    differenceArray[0] = array[0];
+    differenceArray[n] = 0;
+    for (int i = 1; i < n; i++) {
+      differenceArray[i] = array[i] - array[i - 1];
+    }
+  }
+
+  // Does range update
+  public static void update(int[] differenceArray, int l, int r, int x) {
```
[the remark this answers is not in the document]
Comment (AtharvaThorve) [reply src/main/java/com/williamfiset/algorithms/other/RangeUpdate.java]: Yes, I'll add this part now.

**Comment by williamfiset** [comment]
Hi Atharva, I did a few changes to your PR, let me know what you think.

Change Summary:
1. Renamed RangeUpdate class to LazyRangeUpdater
1. Added a test to do random range addition queries
1. Changed the `update` method to be called `add`
1. Added some clarification comments.

If all this looks good I'll go ahead and merge to master

**Comment by AtharvaThorve** [comment]
Yes, everything looks great @williamfiset. Thank you for your feedback and I'll contribute more to this repository as time progresses.

**Comment by williamfiset** [comment]
Done, thanks for the PR!

**Pull request merged**